lib/gen_stage.ex

```diff
@@ -2321,8 +2323,11 @@ defmodule GenStage do
         :ok
 
       excess ->
-        error_msg = 'GenStage producer ~tp has discarded ~tp events from buffer'
-        :error_logger.warning_msg(error_msg, [Utils.self_name(), excess])
+        tags = case Metrics.metrics_self_name() do
```

Comment:
just curious, why not put the logic into GenStage.Metrics and have a function there that returns a list of tags? there's a pattern of that in DiscordSessions.Metrics

Reply:
yeah, that makes sense.
I was hoping there was some idea of "default tags" in Instruments, which I think would be the cleanest place to put this, but since that doesn't exist I just put this logic wherever.
I'll move this into `GenStage.Metrics` to be cleaner!

Reply:
Done